backend/typescript/services/implementations/camperService.ts

```diff
@@ -718,18 +717,50 @@ class CamperService implements ICamperService {
           );
         }
 
-        const oldCampSessionOriginalCampers = oldCampSession.campers; // for roll back
-
-        if (newCampSession) {
-          // campers for new camp session
-          newCampSession.campers = newCampSession.campers.concat(oldCampers);
+        const timeNow = new Date();
+        // Find the earliest date for the camp session
+        const earliestDate = newCampSession.dates.reduce(
+          (earliest, date) => (earliest < date ? earliest : date),
+          new Date(Infinity),
+        );
 
-          // campers for old camp session
-          oldCampSession.campers = oldCampSession.campers.filter(
-            (camperId) => !camperIds.includes(camperId.toString()),
+        const newCampSessionDate = earliestDate;
+        if (timeNow > newCampSessionDate) {
+          throw new Error(
+            `Error: you can only move campers to future sessions`,
           );
         }
 
+        const newCampSessionCapcityRemaining =
+          newCampSession.capacity - newCampSession.campers.length;
+        const oldCampSessionOriginalCampers = oldCampSession.campers; // for roll back
+
+        if (
+          oldCampSessionOriginalCampers.length > newCampSessionCapcityRemaining
+        ) {
+          throw new Error(`Error: camp session to move to is full`);
+        }
+
+        // campers for new camp session
+        newCampSession.campers = newCampSession.campers.concat(oldCampers);
```

Comment:
Looks like you removed the if clause around this and the `oldCampSession.campers...`  lines, was this a neccesary change? If not, we could put it back but just verify once.

Reply:
Nvm, I see we still `have if (!newCampSession)`, which makes the old if statement unneccesary, it's good you removed it.